libkbfs/interfaces.go

```diff
@@ -1697,9 +1697,14 @@ type Observer interface {
 	// updated locally, but not yet saved at the server.
 	LocalChange(ctx context.Context, node Node, write WriteRange)
 	// BatchChanges announces that the nodes have all been updated
-	// together atomically.  Each NodeChange in changes affects the
-	// same top-level folder and branch.
-	BatchChanges(ctx context.Context, changes []NodeChange)
+	// together atomically.  Each NodeChange in `changes` affects the
+	// same top-level folder and branch. `allAffectedNodeIDs` is a
+	// list of all the nodes that had their underlying data changed,
+	// even if it wasn't an user-visible change (e.g., if a
+	// subdirectory was updated, the directory block for the TLF root
+	// is updated but that wouldn't be visible to a user).
```

Comment:
What happens if nodes throughout a hierarchy are modified (as they would be)? It looks like we're going to `Reset` once for each `NodeID` affected.

Reply:
We're only registering the Node corresponding to the top directory of the repo (i.e., `.kbfs_git/reponame`), not any of the child nodes.  git updates happen in single-op batches, so we'll only get one `BatchChanges` callback per update where that node's ID has changed.  Any changes on nodes below should also be included in the same batch, but aren't registered with `AutogitManager`, so they won't cause a `Reset`.

Reply:
Got it, thanks.